src/test/java/de/rabea/server/NetworkStub.java

```diff
@@ -0,0 +1,35 @@
+package de.rabea.server;
+
+import java.io.IOException;
+
+import static de.rabea.TestHelper.asString;
+
+public class NetworkStub implements Connection {
+
+    private final String message;
+    private String exception = "";
+    public String returnedResponse;
+
+    public NetworkStub(String message) {
+        this.message = message;
+    }
+
+    public NetworkStub(String message, String exception) {
+        this.message = message;
+        this.exception = exception;
+    }
+
+    public String read() {
+       return message;
+    }
+
+    public void write(String response, byte[] body) {
+        returnedResponse = response + asString(body);
+    }
+
+    public void close() throws IOException {
+        if (exception.equals("throw exception")) {
```

Comment:
A string comparison seems rather unusual here.
Why not have something more explicit like a named constructor? So you could call `new NetworkStub.thatThrows()` in a test.
The problem with the string is, _I_ as a user of this double now need to know exactly one particular string value to use it in a correct way.